taskipy/task.py

```diff
@@ -1,71 +1,44 @@
-        self.name = task_name
-        self.args = args
-
-    def __str__(self) -> str:
-        return self.name
+    def __init__(self, task_name: str, task_toml_contents: object):
+        self.__task_name = task_name
+        self.__task_command = self.__extract_task_command(task_toml_contents)
+        self.__task_description = self.__extract_task_description(task_toml_contents)
 
     @property
-    def pre_task(self) -> Optional[str]:
-        return self.__find_hooks('pre')
+    def name(self):
+        return self.__task_name
 
     @property
-    def post_task(self) -> Optional[str]:
-        return self.__find_hooks('post')
+    def command(self):
+        return self.__task_command
 
     @property
-    def commands(self) -> List[str]:
-        try:
-            command = self.project.tasks[self.name]
-            commands = []
-            commands.append(
-                ' '.join([command] + [self.__quote_arg(arg) for arg in self.args])
-            )
-
-            if self.pre_task:
-                commands.insert(0, self.pre_task)
+    def description(self):
+        return self.__task_description
 
-            if self.post_task:
-                commands.append(self.post_task)
+    def __extract_task_command(self, task_toml_contents: object) -> str:
+        if isinstance(task_toml_contents, str):
+            return task_toml_contents
 
-            return commands
-        except KeyError:
-            raise TaskNotFoundError(self.name)
-
-    @property
-    def runner(self) -> Optional[str]:
-        try:
-            runner = self.project.settings['runner']
+        if isinstance(task_toml_contents, dict):
+            try:
+                return task_toml_contents['cmd']
+            except KeyError:
+                raise MalformedTaskError(self.__task_name, 'the task item does not have the "cmd" property')
 
-            if not isinstance(runner, str):
-                raise InvalidRunnerTypeError()
+        raise MalformedTaskError(self.__task_name, 'tasks must be strings, or dicts that contain { cmd, help }')
 
-            return runner.strip()
-        except (KeyError, MissingTaskipySettingsSectionError):
-            return None
+    def __extract_task_description(self, task_toml_contents: object) -> str:
+        if isinstance(task_toml_contents, str):
+            return ''
 
-    def __find_hooks(self, hook_type: str) -> Optional[str]:
-        try:
-            return self.project.tasks[f'{hook_type}_{self.name}']
-        except KeyError:
-            return None
+        if isinstance(task_toml_contents, dict):
+            try:
+                return task_toml_contents['help']
+            except KeyError:
+                return ''
 
-    def __quote_arg(self, arg: str) -> str:
-        if platform.system() == 'Windows':
-            return mslex.quote(arg)
-        return shlex.quote(arg)
+        raise MalformedTaskError(self.__task_name, 'tasks must be strings, or dicts that contain { cmd, help }')
```

Comment:
This error message seems to convey that the tasks _must_ contain `cmd` and `help` when using a dict. However, there could just be `cmd`. I also don't know if we want to use the term "dict" for dictionary or "table" since that is what toml calls them (inline tables and standard tables).

Reply:
Nevermind to the table comment:

https://toml.io/en/v1.0.0#table
"Tables (also known as hash tables or dictionaries)..."

Reply:
regarding the first part, how would you convey that `help` is optional, without making the error message too long? because, if I have to choose between not documenting "`help` being optional" and a lengthy error message, I would take the UX path here and go with the first option :)

Reply:
I agree with you. It's just something I figured I'd mention. It may have been better to prefix this as a general comment. I don't think it's a big deal. An option to consider could be to refer to the `{ cmd, help }` as a schema that must be followed, rather than saying contain. Your call whether you think that is better or not.